libcnb-data/src/buildpack.rs

```diff
@@ -1,520 +0,0 @@
-    /// use libcnb_data::buildpack::BuildpackId;
-    ///
-    /// let buildpack_id: BuildpackId = buildpack_id!("heroku/java");
-    /// ```
-    buildpack_id,
-    /// The ID of a buildpack.
-    ///
-    /// It MUST only contain numbers, letters, and the characters `.`, `/`, and `-`.
-    /// It also MUST NOT be `config` or `app`.
-    ///
-    /// Use the [`buildpack_id`](crate::buildpack_id) macro to construct a `BuildpackId` from a
-    /// literal string. To parse a dynamic string into a `BuildpackId`, use
-    /// [`str::parse`](str::parse).
-    ///
-    /// # Examples
-    /// ```
-    /// use libcnb_data::buildpack::BuildpackId;
-    /// use libcnb_data::buildpack_id;
-    ///
-    /// let from_literal = buildpack_id!("heroku/jvm");
-    ///
-    /// let input = "heroku/jvm";
-    /// let from_dynamic: BuildpackId = input.parse().unwrap();
-    /// assert_eq!(from_dynamic, from_literal);
-    ///
-    /// let input = "app";
-    /// let invalid: Result<BuildpackId, _> = input.parse();
-    /// assert!(invalid.is_err());
-    /// ```
-    BuildpackId,
-    BuildpackIdError,
-    r"^(?!app$|config$)[[:alnum:]./-]+$"
-);
-
-libcnb_newtype!(
-    buildpack,
-    /// Construct a [`StackId`] value at compile time.
-    ///
-    /// Passing a string that is not a valid `StackId` value will yield a compilation error.
-    ///
-    /// # Examples:
-    /// ```
-    /// use libcnb_data::stack_id;
-    /// use libcnb_data::buildpack::StackId;
-    ///
-    /// let stack_id: StackId = stack_id!("heroku-20");
-    /// ```
-    stack_id,
-    /// The ID of a stack.
-    ///
-    /// It MUST only contain numbers, letters, and the characters `.`, `/`, and `-`.
-    ///
-    /// Use the [`stack_id`](crate::stack_id) macro to construct a `StackId` from a
-    /// literal string. To parse a dynamic string into a `StackId`, use
-    /// [`str::parse`](str::parse).
-    ///
-    /// # Examples
-    /// ```
-    /// use libcnb_data::buildpack::StackId;
-    /// use libcnb_data::stack_id;
-    ///
-    /// let from_literal = stack_id!("heroku-20");
-    ///
-    /// let input = "heroku-20";
-    /// let from_dynamic: StackId = input.parse().unwrap();
-    /// assert_eq!(from_dynamic, from_literal);
-    ///
-    /// let input = "_stack_";
-    /// let invalid: Result<StackId, _> = input.parse();
-    /// assert!(invalid.is_err());
-    /// ```
-    StackId,
-    StackIdError,
-    r"^[[:alnum:]./-]+$"
-);
-
-#[derive(thiserror::Error, Debug)]
-pub enum BuildpackTomlError {
-    #[error("Found `{0}` but value MUST be in the form `<major>.<minor>` or `<major>` and only contain numbers.")]
-    InvalidBuildpackApi(String),
-
-    #[error("Stack with id `*` MUST NOT contain mixins, however the following mixins were specified: `{}`", .0.join("`, `"))]
-    InvalidAnyStack(Vec<String>),
-
-    #[error("Invalid Stack ID: {0}")]
-    InvalidStackId(#[from] StackIdError),
-
-    #[error("Invalid Buildpack ID: {0}")]
-    InvalidBuildpackId(#[from] BuildpackIdError),
```

Comment:
This `InvalidBuildpackId` variant was unused, so didn't get migrated anywhere, instead removed.